### PR TITLE
Update subpackages

### PR DESCRIPTION
**Pull request by k-dominik**
This PR adds the `spec-bioimage-io` package and allows users to clone submodules with their preferred cloning method :)

~~This is still a draft - some stuff in tiktorch has to be changed to reflect submodule changes/updates~~ surprisingly it worked :o

### CODE REVIEW

**Comment by k-dominik** [comment]
> If we rely on the other repos being cloned by the dev we should document this in the Readme.md

Do you think of something like a setup guide - how to get this repo "rolling" (... so document `git submodule init && git submodule update`)?

**Comment by FynnBe** [comment]
I suppose we should just adapt https://github.com/ilastik/tiktorch/blob/1181ceda2d173285f66f9621457239ef199b07b2/Makefile#L25-L30

**Comment by FynnBe** [comment]
oh well.. that's not enough... maybe add an echo in there reporting that the submodules are expected in `$(ROOT_DIR)/..`?

**Comment by FynnBe** [comment]
maybe I'm misunderstanding the submodules.. what happens if you don't have the repos cloned in advance? and the url is dead?

**Comment by k-dominik** [comment]
so the process is:

```
# clone tiktorch
git clone https://github.com/ilastik/tiktorch

# initialize submodules
git submodule init

# update submodules
git submodule update

# make devenv
make devenv
```

the cloning happens during `git submodule update` and it is always put into `vendor`... this is how it's configured.

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,9 @@
...
 	path = vendor/python-bioimage-io
-	url = git@github.com:bioimage-io/python-bioimage-io.git
+	url = ../../bioimage-io/python-bioimage-io.git
 [submodule "vendor/pytorch-bioimage-io"]
 	path = vendor/pytorch-bioimage-io
-	url = git@github.com:bioimage-io/pytorch-bioimage-io.git
+	url = ../../bioimage-io/pytorch-bioimage-io.git
+[submodule "vendor/spec-bioimage-io"]
+	path = vendor/spec-bioimage-io
+	url = ../../bioimage-io/spec-bioimage-io
```
Comment (FynnBe) [inline .gitmodules:9]: missing `.git` or too manty `.git` up there?

**Comment by FynnBe** [comment]
but this still requires the previous cloning of the submodules to the url detination `root/vendor/../../`, right?

**Comment by k-dominik** [comment]
> but this still requires the previous cloning of the submodules to the url detination `root/vendor/../../`, right?

the steps described are all that's needed. No "manual" cloning. Cloning is happening on `git submodule update`. The relative urls are wrt to the tiktorch repository.

**Review by FynnBe — APPROVED** [review]

**Comment by FynnBe** [comment]
but why do the tests fail with e.g.
```
tiktorch/server/prediction_pipeline/_model_adapters/_model_adapter.py:5: in <module>
    from bioimageio.spec import nodes
E   ModuleNotFoundError: No module named 'bioimageio.spec'
```

**Comment by FynnBe** [comment]
could this be due to cached env?
```
  install_conda_env:
    environment:
      TIKTORCH_ENV_NAME: tiktorch-server-env
      TIKTORCH_ENV_PREFIX: /opt/conda/envs/tiktorch-server-env
    docker:
      - image: continuumio/miniconda3
    working_directory: ~/repo
    steps:
      - restore_cache:
          keys:
            - v1-repo-{{ .Environment.CIRCLE_SHA1 }}
      - restore_cache:
          keys:
            - v7-dependencies-{{ checksum "environment.yml" }}

      - run: conda config --set channel_priority strict
      - run: |
          if [ ! -d ${TIKTORCH_ENV_PREFIX} ]; then
              echo "Creating new environment ${TIKTORCH_ENV_NAME}"
              conda install -c conda-forge conda-build make
              make devenv
          fi
      - save_cache:
          paths:
            - /opt/conda/envs
          key: v7-dependencies-{{ checksum "environment.yml" }}
          ```

**Comment by k-dominik** [comment]
I don't think that's it. As far as I understand caching here, this cache is shared only for a current build... hmmm

**Pull request merged**